### ISSUE TITLE
strange behavior of plot/writeTopology after deleteLeaf!

### ISSUE THREAD

**Issue by crsl4** [crsl4/PhyloNetworks.jl#84]
Strange behavior with `plot` or `writeTopology` after deleting leaves:
```
using PhyloNetworks, PhyloPlots
t = readTopology("(((((Erob01:1.00000050002909e-6,Erob02:1.00000050002909e-6):0.006840603204099063,Bmu:0.004573944171182547):0.000847232768215233,(Bbo02:0.0005092998603560498,Bbo01:9.300098266560527e-5):0.005952818628799544):0.00025736797784936386,(Mno00:0.0054558432758979826,Bph03:0.004523118149092446):0.0012985628624801434):0.000595481858089789,(balMys:0.0022409920465227616,Egl00:0.0030752120031449152):0.008484326240849983,Bac00:0.008821956742683938);")
PhyloNetworks.deleteLeaf!(t,"balMys")
PhyloNetworks.deleteLeaf!(t,"Egl00")
plot(t,:R)
```
The resulting plot (attached: test.pdf) looks weird. However, if you look at the HybridNetwork object `t` (we need to draw the network by hand), it seems to be correct:
```
julia> printEdges(t)
edge parent child  length  hybrid isMajor gamma   containRoot inCycle istIdentitiable
1    1      -6     0.000   false  true    1       true        -1      false
2    -6     2      0.000   false  true    1       true        -1      false
3    -6     -5     0.007   false  true    1       true        -1      true 
4    -5     3      0.005   false  true    1       true        -1      false
5    -5     -4     0.001   false  true    1       true        -1      true 
6    -7     4      0.001   false  true    1       true        -1      false
7    -7     5      0.000   false  true    1       true        -1      false
8    -4     -7     0.006   false  true    1       true        -1      true 
9    -4     -3     0.000   false  true    1       true        -1      true 
10   -8     6      0.005   false  true    1       true        -1      false
11   -8     7      0.005   false  true    1       true        -1      false
12   -3     -8     0.001   false  true    1       true        -1      true 
17   -3     10     0.009   false  true    1       true        -1      false

julia> printNodes(t)
node leaf  hybrid hasHybEdge name   inCycle edges'numbers
1    true  false  false      Erob01 -1      1   
2    true  false  false      Erob02 -1      2   
-6   false false  false             -1      1    2    3   
3    true  false  false      Bmu    -1      4   
-5   false false  false             -1      3    4    5   
4    true  false  false      Bbo02  -1      6   
5    true  false  false      Bbo01  -1      7   
-7   false false  false             -1      6    7    8   
-4   false false  false             -1      5    8    9   
6    true  false  false      Mno00  -1      10  
7    true  false  false      Bph03  -1      11  
-8   false false  false             -1      10   11   12  
-3   false false  false             -1      9    12   17  
10   true  false  false      Bac00  -1      17  
```
Also, when we try to write in parenthetical format, the result is also wrong (leaf Erob01 disappears):
```
julia> writeTopology(t)
"((Erob02:1.00000050002909e-6,(Bmu:0.004573944171182547,((Bbo02:0.0005092998603560498,Bbo01:9.300098266560527e-5):0.005952818628799544,((Mno00:0.0054558432758979826,Bph03:0.004523118149092446):0.0012985628624801434,Bac00:0.009417438600773726):0.00025736797784936386):0.000847232768215233):0.006840603204099063):1.00000050002909e-6);"

julia> t
PhyloNetworks.HybridNetwork, Rooted Network
13 edges
14 nodes: 8 tips, 0 hybrid nodes, 6 internal tree nodes.
tip labels: Erob01, Erob02, Bmu, Bbo02, ...
((Erob02:0.0,(Bmu:0.005,((Bbo02:0.001,Bbo01:0.0):0.006,((Mno00:0.005,Bph03:0.005):0.001,Bac00:0.009):0.0):0.001):0.007):0.0);
```

The problem appears to be the root (node=1 which is the leaf Erob01). If we change the root, we can now write and plot correctly.
```
julia> t.root
1

julia> rootatnode!(t,-6)
PhyloNetworks.HybridNetwork, Rooted Network
13 edges
14 nodes: 8 tips, 0 hybrid nodes, 6 internal tree nodes.
tip labels: Erob01, Erob02, Bmu, Bbo02, ...
(Erob01:0.0,Erob02:0.0,(Bmu:0.005,((Bbo02:0.001,Bbo01:0.0):0.006,((Mno00:0.005,Bph03:0.005):0.001,Bac00:0.009):0.0):0.001):0.007);
```
Even if we try to change back the root at 1, we now get a warning, and correct parenthetical format.
```
julia> rootatnode!(t,1)
INFO: node 1 is a leaf. Will create a new node if needed, to set taxon "Erob01" as outgroup.
PhyloNetworks.HybridNetwork, Rooted Network
14 edges
15 nodes: 8 tips, 0 hybrid nodes, 7 internal tree nodes.
tip labels: Erob01, Erob02, Bmu, Bbo02, ...
(Erob01:0.0,(Erob02:0.0,(Bmu:0.005,((Bbo02:0.001,Bbo01:0.0):0.006,((Mno00:0.005,Bph03:0.005):0.001,Bac00:0.009):0.0):0.001):0.007):0.0);
```

I know that `deleteLeaf!` is mostly an internal function for `snaq`, but should we add a warning somewhere in case someone wants to use it? Or maybe we can have an exported function to delete leaves that prevents the root from being at a leaf?
[test.pdf](https://github.com/crsl4/PhyloNetworks.jl/files/2320002/test.pdf)

**Comment by cecileane** [comment]
What do you get when you use `deleteleaf!` instead of `deleteLeaf!`? The second form should be internal (not exported), and should not be used: it does a few things that could cause problems, as documented in the docstring. Could that be the problem?

**Comment by crsl4** [comment]
I had forgotten about `deleteLeaf!` and `deleteleaf!`. I get the same strange behavior though when I use `deleteleaf!`. 
```
using PhyloNetworks, PhyloPlots
t = readTopology("(((((Erob01:1.00000050002909e-6,Erob02:1.00000050002909e-6):0.006840603204099063,Bmu:0.004573944171182547):0.000847232768215233,(Bbo02:0.0005092998603560498,Bbo01:9.300098266560527e-5):0.005952818628799544):0.00025736797784936386,(Mno00:0.0054558432758979826,Bph03:0.004523118149092446):0.0012985628624801434):0.000595481858089789,(balMys:0.0022409920465227616,Egl00:0.0030752120031449152):0.008484326240849983,Bac00:0.008821956742683938);")
deleteleaf!(t,"balMys")
deleteleaf!(t,"Egl00")
plot(t,:R)
```
This seems to be related to the root left at a leaf. If I manually change the root to a leaf node (even without `deleteleaf!` or `deleteLeaf!`), I also get the same behavior:
```
using PhyloNetworks, PhyloPlots
t = readTopology("(((((Erob01:1.00000050002909e-6,Erob02:1.00000050002909e-6):0.006840603204099063,Bmu:0.004573944171182547):0.000847232768215233,(Bbo02:0.0005092998603560498,Bbo01:9.300098266560527e-5):0.005952818628799544):0.00025736797784936386,(Mno00:0.0054558432758979826,Bph03:0.004523118149092446):0.0012985628624801434):0.000595481858089789,(balMys:0.0022409920465227616,Egl00:0.0030752120031449152):0.008484326240849983,Bac00:0.008821956742683938);")
t.root = 1
plot(t,:R)
```
I know no one will do this, but maybe we want `deleteleaf!` to leave the root in a non-leaf node to prevent this issue?

**Comment by cecileane** [comment]
This is fixed in 7cf95a556ce38e9fb71559d4da23ece1f063ebcf .
`deleteleaf!` now leaves the root node as is, even if that node becomes of degree 2. That's the default behavior. With option `unroot=true`, the root node will be deleted if it becomes of degree 2 (to avoid *all* degree-2 nodes after leaf deletion).

If the root is manually set to a leaf, then the plotting issue remains. But `deleteleaf!` should not be causing the re-assignment of the root to a leaf.
